client-app/core/api/graphql/utils.ts

```diff
@@ -12,6 +14,19 @@ export function filterActiveQueryNames<TCacheShape = any>(
   return intersection(activeQueryNames, queryNames);
 }
 
+export function toOptimisticResponse<T extends { id?: string } | undefined>(
+  data: T,
+  __typename: string,
+): T | undefined {
+  if (isDefined(data)) {
+    return {
+      ...data,
+      id: data.id ?? uuidv4(),
```

Comment:
do you really need generate id here?

Reply:
There are 3 stages:
1. Already loaded
2. Run mutation, but don't get server response
3. Get server response
optimisticResponse is overwriting state only at stage 2 and revert changes (on error) or apply response (if success) at stage 3. This ID generation is only for stage 2.

Reply:
Renamed, removed unnecessary